convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,122 @@
+    )
+    if not option_present:
+        logger.debug(
+            "Couldn't find CleanupModulesOnExit in firewalld.conf. Treating it as enabled because of default behavior."
+        )
+        return True
+
+    # If the config file has this option set to true/yes, then we need to
+    # return True to ask the user to change it to False.
+    if list(filter(CLEANUP_MODULES_ON_EXIT_REGEX.match, contents)):
+        logger.debug("CleanupModulesOnExit option enabled in %s" % FIREWALLD_CONFIG_FILE)
+        return True
+
+    # Default to return False as it is possible that the CleanupModulesOnExit
+    # is set to no in the config already.
+    logger.debug("CleanupModulesOnExit option is disabled in %s" % FIREWALLD_CONFIG_FILE)
+    return False
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id == "oracle" and system_info.version.major == 8 and system_info.version.minor >= 8:
+            # If firewalld is not present on the system, we can just skip skip.
+            if not system_info.is_rpm_installed("firewalld"):
+                logger.info("The firewalld package is not installed. Nothing to do.")
+                return
+
+            if not systeminfo.is_systemd_managed_service_running("firewalld"):
+                logger.info("Firewalld service reported that it is not running.")
+                return
+
+            if _is_modules_cleanup_enabled():
+                self.set_result(
+                    level="ERROR",
+                    id="FIREWALLD_MODULES_CLEANUP_ON_EXIT_CONFIG",
+                    title="Firewalld is set to cleanup modules after exit.",
+                    description="Firewalld running on Oracle Linux 8 can lead to a conversion failure.",
+                    diagnosis=(
+                        "We've detected that firewalld unit is running and that causes iptables and nftables "
+                        "failures on Oracle Linux 8 and under certain conditions it can lead to a conversion failure."
+                    ),
+                    remediation=(
+                        "Set the option CleanupModulesOnExit in /etc/firewalld/firewalld.conf "
+                        "to no prior to running convert2rhel:\n"
+                        " sed -i -- 's/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g' /etc/firewalld/firewalld.conf\n && firewall-cmd --reload."
+                        " You can change the option back to yes after the system reboot "
+                        "- that is after the system boots into the RHEL kernel."
+                    ),
+                )
+                return
+
+        logger.info("Skipping the check as it is relevant only for Oracle Linux 8.8 and above.")
```

Comment:
Not a PR blocker, but: If none of the above condition is true, then this message is printed. 

Scenario:
* we are on oracle8.8
* the firewalld is running
* the config file contains the `CleanupModulesOnExit=no`
then all the ifs are skipped and no default `return` is present, so the `logger.info("Skipping the check as it is relevant only for Oracle Linux 8.8 and above.")` is printed even though we are on 8.8 :D

```
14:28:50             out: [2023-10-30T14:28:50+0000] TASK - [Prepare: Gather system information] *******************************
14:28:50             out: [2023-10-30T14:28:50+0000] DEBUG - Calling command 'uname -i'
14:28:50             out: Skipping the execution of 'rpm -Va'.
14:28:50             out: [2023-10-30T14:28:50+0000] DEBUG - Calling command 'uname -r'
14:28:50             out: [2023-10-30T14:28:50+0000] DEBUG - Booted kernel VRA (version, release, architecture): 4.18.0-477.27.1.el8_8.x86_64
14:28:50             out: Checking internet connectivity using address 'https://static.redhat.com/test/rhel-networkmanager.txt'.
14:28:50             out: Successfully connected to address 'https://static.redhat.com/test/rhel-networkmanager.txt', internet connection seems to be available.
14:28:50             out: [2023-10-30T14:28:50+0000] DEBUG - Calling command '/usr/bin/systemctl show -p ActiveState dbus'
14:28:50             out: Name:                Oracle Linux Server
14:28:50             out: OS version:          8.8
14:28:50             out: Architecture:        x86_64
14:28:50             out: Config filename:     oracle-8-x86_64.cfg
14:28:53             out: 
14:28:53             out: [2023-10-30T14:28:53+0000] TASK - [Prepare: Clear YUM/DNF version locks] *****************************
14:28:53             out: Usage of YUM/DNF versionlock plugin not detected.
14:28:53             out: 
14:28:53             out: [2023-10-30T14:28:53+0000] TASK - [Prepare: Clean yum cache metadata] ********************************
14:28:53             out: [2023-10-30T14:28:53+0000] DEBUG - Calling command 'yum clean metadata --enablerepo=* --quiet'
14:28:53             out: [2023-10-30T14:28:53+0000] DEBUG - Output of yum clean metadata:
14:28:53             out: 
14:28:53             out: Cached repositories metadata cleaned successfully.
14:28:53             out: 
14:28:53             out: [2023-10-30T14:28:53+0000] TASK - [Prepare: Check whether system is ready for conversion] ************
14:28:53             out: 
14:28:53             out: [2023-10-30T14:28:53+0000] TASK - [Prepare: Check that firewalld is running] *************************
14:28:53             out: [2023-10-30T14:28:53+0000] DEBUG - Calling command '/usr/bin/systemctl show -p ActiveState firewalld'
14:28:53             out: [2023-10-30T14:28:53+0000] DEBUG - CleanupModulesOnExit option is disabled in /etc/firewalld/firewalld.conf
14:28:53             out: Skipping the check as it is relevant only for Oracle Linux 8.8 and above.
14:28:53             out: CHECK_FIREWALLD_AVAILABILITY has succeeded
```
http://artifacts.osci.redhat.com/testing-farm/cfa41dc4-a364-4e38-9237-b41e280da9dc/work-rhsmf6bxgaaj/log.txt